site/source/docs/porting/Debugging.rst

```diff
@@ -137,6 +137,45 @@ Debug printouts can even execute arbitrary JavaScript. For example::
   }
 
 
+Printing exception messages
+===========================
+
+Exceptions are thrown from WebAssembly using exception pointers, which means that try/catch/finally blocks in JavaScript will only receive a number, which represents a pointer into linear memory. In order to get the exception message, the user will need to create some WASM code which will extract the meaning from the exception, for example:
+
+.. code-block:: cpp
+
+  #include <bind.h>
+
+  std::string getExceptionMessage(int exceptionPtr) {
+    return std::string(reinterpret_cast<std::exception *>(exceptionPtr)->what());
+  }
+
+  EMSCRIPTEN_BINDINGS(Bindings) {
+    emscripten::function("getExceptionMessage", &getExceptionMessage);
+  };
+
+And then handling the exceptions in Javascript will look like this:
+
+.. code-block:: javascript
+
+  try {
+    ... // some code that calls WebAssembly
+  } catch (exception) {
+    console.error(Module.getExceptionMessage(exception));
+  } finally {
+    ...
+  }
+
+It's important to notice that this code will work only for thrown statically allocated exceptions. If your code throws other objects, such as strings or dynamically allocated exceptions, the handling code will need to take that into account. For example, in order to handle thrown strings use:
```

Comment:
This is a little confusing because in your example it is a string that is thrown and std::string is dynamically allocated.  

Perhaps  something like:  `this code will work only for throwing char* pointers.    If you throw other types your `getExceptionMessage` might need to do different things.`

Reply:
If you were confused then the description is confusing, I'm not gonna argue with that. 
The thing is that I didn't give an example in which a C++ string is thrown, so I'd like to understand what's the source of the confusion. Can you please explain what made you think that `std::string` is thrown?

Reply:
I was confused by the because getExceptionMessage returns a std::string which is dynamically allocated.

I think I understand better now.   How about something like `in this example the thing being thrown in char*.    The `getExceptionMessage` function would  different if some other C/C++ object was thrown`.

Reply:
I rephrased the docs. Are they clearer now?